File: jinyun/web/src/main/java/com/jinyun/web/controller/LoginController.java
package com.jinyun.web.controller;

import com.alibaba.fastjson.JSONObject;
import com.jinyun.web.annotation.UserLoginToken;
import com.jinyun.web.entity.User;
import com.jinyun.web.service.TokenService;
import com.jinyun.web.service.UserService;
import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.multipart.MultipartFile;

import java.io.File;
import java.io.IOException;

@Api(value = "desc of class", tags="Login",description = "登录")
@RestController
@RequestMapping("login")
public class LoginController {
    @Autowired
    UserService userService;
    @Autowired
    TokenService tokenService;
    //登录
    @PostMapping("/login")
    public Object login(@RequestBody User user){
        JSONObject jsonObject=new JSONObject();
        User userForBase=userService.findByUsername(user);
        if(userForBase==null){
            jsonObject.put("message","登录失败,用户不存在");
            return jsonObject;
        }else {
            if (!userForBase.getPassword().equals(user.getPassword())){
                jsonObject.put("message","登录失败,密码错误");
                return jsonObject;
            }else {
                String token = tokenService.getToken(userForBase);
                jsonObject.put("token", token);
                jsonObject.put("user", userForBase);
                return jsonObject;
            }
        }
    }

    @UserLoginToken
    @GetMapping("/getMessage")
    public String getMessage(){
        return "你已通过验证";
    }

}

File: jinyun/web/src/main/kotlin/com/jinyin/web/LoginController.kt
package com.jinyin.web

import com.alibaba.fastjson.JSONObject
import com.jinyun.web.annotation.UserLoginToken
import com.jinyun.web.entity.User
import com.jinyun.web.service.TokenService
import com.jinyun.web.service.UserService
import io.swagger.annotations.Api
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@Api(value = "desc of class", tags = arrayOf("Login1"), description = "登录")
@RestController
@RequestMapping("login1")
class LoginController {
    @Autowired
    var userService: UserService? = null
    @Autowired
    var tokenService: TokenService? = null

    //登录
    @PostMapping("/login1")
    fun login(user: User): Any {
        val jsonObject = JSONObject()
        val userForBase = userService!!.findByUsername(user)
        return if (userForBase == null) {
            jsonObject["message"] = "登录失败,用户不存在"
            jsonObject
        } else {
//            if (userForBase.password != user.password) {
//                jsonObject["message"] = "登录失败,密码错误"
//                jsonObject
//            } else {
//                val token = tokenService!!.getToken(userForBase)
//                jsonObject["token"] = token
//                jsonObject["user"] = userForBase
//                jsonObject
//            }
        }
    }

    @get:GetMapping("/getMessage1")
    @get:UserLoginToken
    val message: String
        get() = "你已通过验证"
}